dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/EditDatafilesPage.java

```diff
@@ -787,33 +799,18 @@ public String save() {
             Map<Long, String> deleteStorageLocations = null;
 
             if (!filesToBeDeleted.isEmpty()) {
-                deleteStorageLocations = datafileService.getPhysicalFilesToDelete(filesToBeDeleted);
+                deleteStorageLocations = datafileDao.getPhysicalFilesToDelete(filesToBeDeleted);
             }
 
-            Command<Dataset> cmd;
-            try {
-                cmd = new UpdateDatasetVersionCommand(dataset, dvRequestService.getDataverseRequest(), filesToBeDeleted, clone);
-                ((UpdateDatasetVersionCommand) cmd).setValidateLenient(true);
-                dataset = commandEngine.submit(cmd);
-
-            } catch (EJBException ex) {
-                StringBuilder error = new StringBuilder();
-                error.append(ex).append(" ");
-                error.append(ex.getMessage()).append(" ");
-                Throwable cause = ex;
-                while (cause.getCause() != null) {
-                    cause = cause.getCause();
-                    error.append(cause).append(" ");
-                    error.append(cause.getMessage()).append(" ");
-                }
-                logger.log(Level.INFO, "Couldn''t save dataset: {0}", error.toString());
-                populateDatasetUpdateFailureMessage();
-                return null;
-            } catch (CommandException ex) {
-                //FacesContext.getCurrentInstance().addMessage(null, new FacesMessage(FacesMessage.SEVERITY_ERROR, "Dataset Save Failed", " - " + ex.toString()));
-                logger.log(Level.INFO, "Couldn''t save dataset: {0}", ex.getMessage());
-                populateDatasetUpdateFailureMessage();
-                return null;
+            Try<Dataset> updateDatasetOperation = Try.of(() -> datasetVersionService.updateDatasetVersion(workingVersion, true))
+                    .onSuccess(updatedDataset -> dataset = updatedDataset)
```

Comment:
Why? Personally i like functional structure :P.

Reply:
1. It would increase usability of `updateDatasetOperation`
2. I don't fancy setting class fields in `.onSuccess/Failure` closes :P 
But it's kinda fine with me.

Reply:
1. Why do you mean by usability? One way is just functional and the other is OOP(? i think there is other name for that) but they are both working the same way.
2. Yeah it is kinda specific to JSF, usually we would probably go for safer immutable options so you would never have to set it. It would probably be something like ```.onSuccess(datasetMapper::mapToDto)```

Reply:
By usability i mean that we don't just create `updateDatasetOperation` to make `isFailure()` check and if so return.
But as I said: it's not a big deal for me.